### DISCUSSION TITLE
A lumped element dealing with damping and diffusion

### DISCUSSION THREAD

**Discussion by chaohc** [atcollab/at#549] [Ideas]
Hello everyone,

I need to fix the longitudinal motion with ID. A simple fix is to use a single lumped element every turn that changes energy from both the classical radiation and incoherent synchrotron radiation, similar to SReffects in elegant. The new element must be able to overwrite the old damping partitions and add the additional energy loss. These values will be derived from ID with hard edge model or other analytical formulas. Ideally I hope any arbitrary values can be assigned. Of course all other local radiations must be turn off during the tracking.

Is it possible to have such a handy element?

Bests,
Hung-Chun

**Comment by swhite2401** [comment]
Hello,
Elegant elements have been ported to AT in the past. The c source can be integrated in a passmethod to do so, a critical point is to properly convert coordinates from the elegant formalism to the one used in AT.
